### PR TITLE
doc: clarify full Xcode download is not needed

### PR DESCRIPTION
**Pull request by Sjors**
Some people keep thinking that you need to download the full 4GB of Xcode before running `xcode-select`. However `xcode-select` is already present on any freshly installed macOS.

### CODE REVIEW

**Comment by grdddj** [comment]
LGTM!

File: doc/build-osx.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ macOS comes with a built-in Terminal located in:
 
 The Xcode Command Line Tools are a collection of build tools for macOS.
 These tools must be installed in order to build Bitcoin Core from source.
+You do not need to install all of Xcode.
```
Comment (brunoerg) [inline doc/build-osx.md:20]: Seems weird to say "you don't need to install" and the next line "To install, run...". If we want to advertise something, I prefer something like: "Recent MacOS versions may contain Xcode by default"

**Comment by Sjors** [comment]
Some background, it should indeed not be necessary to install Xcode: https://www.freecodecamp.org/news/install-xcode-command-line-tools/

I improved the prose a bit.

File: doc/build-osx.md (hunk 1 above)
Comment (Sjors) [reply doc/build-osx.md]: macOS does not ship with all of Xcode, but they do ship with the `xcode-select` command.

**Comment by fanquake** [comment]
> Some background, it should indeed not be necessary to install Xcode:

It hasn't been required for at least 7 years, that's why we removed the instructions to install in: https://github.com/bitcoin/bitcoin/commit/2692e1b10bd7a0be644ed8a69c54152bc741e855.

**Comment by fanquake** [comment]
> Not sure if this is needed? Anyone to copy-paste the command shouldn't run into issues and if you do random other stuff, you are on your own anyway?

Agree with the above. Not sure where the confusion is coming from, and, [as mentioned](https://github.com/bitcoin/bitcoin/pull/27835#issuecomment-1589083587), this hasn't be required for at least 7 years, if not longer than that. Anyone who is following this guide, and just copy-pasting commands, shouldn't be accidently going to the macOS App Store and downloading the ~8GB Xcode app bundle.

**Pull request closed without merging**